packages/nuxt/src/vite/sourceMaps.ts

```diff
@@ -87,25 +106,189 @@ export function getPluginOptions(
+        logSentryEnablesSourcemap('sourcemap.server', 'hidden');
+        previousUserSourceMapSetting = { client: 'unset', server: 'unset' };
+        break;
+    }
+  } else {
+    if (nuxtSourcemap.client === false) {
+      warnExplicitlyDisabledSourcemap('sourcemap.client');
+      previousUserSourceMapSetting.client = 'disabled';
+    } else if (['hidden', true].includes(nuxtSourcemap.client)) {
+      logKeepSourcemapSetting(sentryModuleOptions, 'sourcemap.client', nuxtSourcemap.client.toString());
+      previousUserSourceMapSetting.client = 'enabled';
+    } else {
+      nuxt.options.sourcemap.client = 'hidden';
+      logSentryEnablesSourcemap('sourcemap.client', 'hidden');
+      previousUserSourceMapSetting.client = 'unset';
+    }
+
+    if (nuxtSourcemap.server === false) {
+      warnExplicitlyDisabledSourcemap('sourcemap.server');
+      previousUserSourceMapSetting.server = 'disabled';
+    } else if (['hidden', true].includes(nuxtSourcemap.server)) {
+      logKeepSourcemapSetting(sentryModuleOptions, 'sourcemap.server', nuxtSourcemap.server.toString());
+      previousUserSourceMapSetting.server = 'enabled';
+    } else {
+      nuxt.options.sourcemap.server = 'hidden';
+      logSentryEnablesSourcemap('sourcemap.server', 'hidden');
+      previousUserSourceMapSetting.server = 'unset';
+    }
+  }
+
+  return previousUserSourceMapSetting;
+}
+
+/** only exported for testing */
+export function changeViteSourcemapSettings(
+  viteConfig: { build?: { sourcemap?: boolean | 'inline' | 'hidden' } },
+  sentryModuleOptions: SentryNuxtModuleOptions,
+): UserSourcemapSetting {
+  viteConfig.build = viteConfig.build || {};
+  const viteSourcemap = viteConfig.build.sourcemap;
+
+  let previousUserSourceMapSetting: UserSourcemapSetting;
+
+  if (viteSourcemap === false) {
+    warnExplicitlyDisabledSourcemap('vite.build.sourcemap');
+    previousUserSourceMapSetting = 'disabled';
+  } else if (viteSourcemap && ['hidden', 'inline', true].includes(viteSourcemap)) {
+    logKeepSourcemapSetting(sentryModuleOptions, 'vite.build.sourcemap', viteSourcemap.toString());
+    previousUserSourceMapSetting = 'enabled';
+  } else {
+    viteConfig.build.sourcemap = 'hidden';
+    logSentryEnablesSourcemap('vite.build.sourcemap', 'hidden');
+    previousUserSourceMapSetting = 'unset';
+  }
+
+  return previousUserSourceMapSetting;
+}
+
+/** only exported for testing */
+export function changeRollupSourcemapSettings(
+  nitroConfig: {
+    rollupConfig?: {
+      output?: {
+        sourcemap?: OutputOptions['sourcemap'];
+        sourcemapExcludeSources?: OutputOptions['sourcemapExcludeSources'];
+      };
+    };
+  },
+  sentryModuleOptions: SentryNuxtModuleOptions,
+): UserSourcemapSetting {
+  nitroConfig.rollupConfig = nitroConfig.rollupConfig || {};
+  nitroConfig.rollupConfig.output = nitroConfig.rollupConfig.output || { sourcemap: undefined };
+
+  let previousUserSourceMapSetting: UserSourcemapSetting;
```

Comment:
l: some vars use `sourcemap` camel-cased, others not

Reply:
Changed all variables/comments to "source maps" and `sourceMap` according to https://tc39.es/source-map/